### PR TITLE
Port mapc

### LINKED ISSUE #916: Port mapc
``` c

DEFUN ("mapc", Fmapc, Smapc, 2, 2, 0,
       doc: /* Apply FUNCTION to each element of SEQUENCE for side effects only.
Unlike `mapcar', don't accumulate the results.  Return SEQUENCE.
SEQUENCE may be a list, a vector, a bool-vector, or a string.  */)
  (Lisp_Object function, Lisp_Object sequence)
{
  register EMACS_INT leni;

  leni = XFASTINT (Flength (sequence));
  if (CHAR_TABLE_P (sequence))
    wrong_type_argument (Qlistp, sequence);
  mapcar1 (leni, 0, function, sequence);

  return sequence;
}
```

### PR DESCRIPTION
**Pull request by krobelus**
Fixes #916

### CODE REVIEW

File: rust_src/src/fns.rs [hunk 1]
```diff
@@ -90,6 +94,16 @@ pub fn quote(args: LispCons) -> LispObject {
     args.car()
 }
 
+#[lisp_fn]
+pub fn mapc(function: LispObject, sequence: LispObject) -> LispObject {
+    let leni = length(sequence) as EmacsInt;
+    if unsafe { CHAR_TABLE_P(sequence) } {
```
Comment (shaleh) [inline rust_src/src/fns.rs]:
```suggestion
    if sequence.is_char_table() {
```

Comment (shaleh) [reply rust_src/src/fns.rs]: We prefer to use methods where possible instead of porting the C macros.

Comment (krobelus) [reply rust_src/src/fns.rs]: thanks, I missed that one!

File: rust_src/src/fns.rs [hunk 2]
```diff
@@ -90,6 +93,16 @@ pub fn quote(args: LispCons) -> LispObject {
     args.car()
 }
 
+#[lisp_fn]
+pub fn mapc(function: LispObject, sequence: LispObject) -> LispObject {
+    let leni = length(sequence) as EmacsInt;
```
Comment (krobelus) [inline rust_src/src/fns.rs]: not sure if `leni` as name is ok
same for the cast but that's a simple solution for now

Comment (shaleh) [reply rust_src/src/fns.rs]: Both are fine.

**Pull request merged**